benchmarks/benchmarks/signal_filtering.py

```diff
@@ -102,3 +102,16 @@ def time_medfilt2d(self, threads):
 
     def peakmem_medfilt2d(self, threads):
         self._medfilt2d(threads)
+
+
+class FreqzRfft(Benchmark):
+    param_names = ['worN']
+    params = [
+        [64, 65, 128, 129, 256, 257, 512, 513],
```

Comment:
With the removal of `10000` below we can use larger numbers here. 65537 is nice in particular because it's prime so should be a worst case for FFT IIRC. Then also include 65538 so that when we hit the branch `n_fft = 2 * (N - 1)` we have two times a prime, which should also be slow for FFT. Same motivation for adding 258.
```suggestion
        [64, 65, 128, 129, 256, 257, 258, 512, 513, 65536, 65537, 65538],
```
FWIW I expect all of these to be as fast or faster than on `main` with your changes, but it's better to have these in here to double check!

Reply:
See my comment above. The benchmark on main for the big numbers timeouts :(. If you think that we don't need to benchmark the vectorized version at all I'll just remove it.

Reply:
I think we can get away without benchmarking the vectorized version, especially if by not doing so we get to test these other, hopefully more informative cases

Reply:
DONE

Here is what I get:

```
main:

[50.00%] ··· signal_filtering.FreqzRfft.time_freqz                                                                                                                         ok
[50.00%] ··· ======= ========= ======= ==========
              whole   nyquist    worN            
             ------- --------- ------- ----------
              False    False      64    1.20±0ms 
              False    False      65    623±0μs  
              False    False     128    722±0μs  
              False    False     129    741±0μs  
              False    False     256    665±0μs  
              False    False     257    1.09±0ms 
              False    False     258    1.08±0ms 
              False    False     512    693±0μs  
              False    False     513    2.01±0ms 
              False    False    65536   5.36±0ms 
              False    False    65537   6.77±0s  
              False    False    65538   7.28±0s  
              False     True      64    722±0μs  
              False     True      65    644±0μs  
              False     True     128    728±0μs  
              False     True     129    775±0μs  
              False     True     256    726±0μs  
              False     True     257    1.13±0ms 
              False     True     258    1.17±0ms 
              False     True     512    744±0μs  
              False     True     513    2.08±0ms 
              False     True    65536   5.68±0ms 
              False     True    65537   7.70±0s  
              False     True    65538   7.84±0s  
               True    False      64    746±0μs  
               True    False      65    623±0μs  
               True    False     128    762±0μs  
               True    False     129    776±0μs  
               True    False     256    737±0μs  
               True    False     257    1.14±0ms 
               True    False     258    1.13±0ms 
               True    False     512    761±0μs  
               True    False     513    2.14±0ms 
               True    False    65536   3.71±0ms 
               True    False    65537   7.30±0s  
               True    False    65538   7.31±0s  
               True     True      64    707±0μs  
               True     True      65    589±0μs  
               True     True     128    695±0μs  
               True     True     129    782±0μs  
               True     True     256    683±0μs  
               True     True     257    1.10±0ms 
               True     True     258    1.10±0ms 
               True     True     512    695±0μs  
               True     True     513    2.06±0ms 
               True     True    65536   3.59±0ms 
               True     True    65537   7.02±0s  
               True     True    65538   7.05±0s  
             ======= ========= ======= ==========

with the fix:

[50.00%] ··· signal_filtering.FreqzRfft.time_freqz                                    
[50.00%] ··· ======= ========= ======= ==========
              whole   nyquist    worN            
             ------- --------- ------- ----------
              False    False      64    9.46±0ms 
              False    False      65    1.78±0ms 
              False    False     128    713±0μs  
              False    False     129    680±0μs  
              False    False     256    674±0μs  
              False    False     257    708±0μs  
              False    False     258    675±0μs  
              False    False     512    688±0μs  
              False    False     513    694±0μs  
              False    False    65536   6.18±0ms 
              False    False    65537   26.7±0ms 
              False    False    65538   12.7±0ms 
              False     True      64    674±0μs  
              False     True      65    668±0μs  
              False     True     128    682±0μs  
              False     True     129    672±0μs  
              False     True     256    679±0μs  
              False     True     257    673±0μs  
              False     True     258    716±0μs  
              False     True     512    707±0μs  
              False     True     513    694±0μs  
              False     True    65536   11.3±0ms 
              False     True    65537   5.32±0ms 
              False     True    65538   26.6±0ms 
               True    False      64    706±0μs  
               True    False      65    689±0μs  
               True    False     128    697±0μs  
               True    False     129    695±0μs  
               True    False     256    693±0μs  
               True    False     257    713±0μs  
               True    False     258    691±0μs  
               True    False     512    694±0μs  
               True    False     513    692±0μs  
               True    False    65536   3.49±0ms 
               True    False    65537   15.0±0ms 
               True    False    65538   7.42±0ms 
               True     True      64    742±0μs  
               True     True      65    744±0μs  
               True     True     128    748±0μs  
               True     True     129    741±0μs  
               True     True     256    734±0μs  
               True     True     257    799±0μs  
               True     True     258    743±0μs  
               True     True     512    748±0μs  
               True     True     513    744±0μs  
               True     True    65536   3.64±0ms 
               True     True    65537   16.0±0ms 
               True     True    65538   7.49±0ms 
             ======= ========= ======= ==========

```